hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -416,38 +416,47 @@ object Simplify {
     case InsertFields(SelectFields(old, _), Seq(), Some(insertFieldOrder)) =>
       SelectFields(old, insertFieldOrder)
 
-    case Let(Seq(), body) =>
+    case Block(Seq(), body) =>
       body
 
-    case Let(xs, Let(ys, body)) =>
-      Let(xs ++ ys, body)
+    case Block(xs, Block(ys, body)) =>
+      Block(xs ++ ys, body)
 
     // assumes `NormalizeNames` has been run before this.
-    case Let(Let.Nested(before, after), body) =>
-      def numBindings(b: (String, IR)): Int =
-        b._2 match {
-          case let: Let => 1 + let.bindings.length
+    case Block(bindings, body) if bindings.exists(_.value.isInstanceOf[Block]) =>
```

Comment:
The point of `Nested` was to not have to repeat the work of finding the first nested `Block` within this one. For some benchmarks (like `matrix_multi_write_nothing`), repeatedly iterating over very long arrays of bindings starts to show up on the profiler.

Reply:
I see your point, I'll restore `Nested`

Reply:
Though I think the fact tha `Nested` copies the bindings array into two new ones defeats any benefit from not having to scan the front of the array again. Maybe `Nested` should just bind the index of the first match.